### PR TITLE
Misc small fixes and improvements

### PR DESCRIPTION
**Pull request by mvieth**
Please see commit messages for details.
Related: https://github.com/PointCloudLibrary/pcl/issues/6178
Fixes https://github.com/PointCloudLibrary/pcl/issues/6162

### CODE REVIEW

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp [hunk 1]
```diff
@@ -451,7 +451,7 @@ pcl::SampleConsensusModelEllipse3D<PointT>::optimizeModelCoefficients (
   OptimizationFunctor functor(this, inliers);
   Eigen::NumericalDiff<OptimizationFunctor> num_diff(functor);
   Eigen::LevenbergMarquardt<Eigen::NumericalDiff<OptimizationFunctor>, double> lm(num_diff);
-  Eigen::VectorXd coeff;
+  Eigen::VectorXd coeff = model_coefficients.cast<double>();
   int info = lm.minimize(coeff);
```
Comment (larshg) [inline sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp:455]:
```suggestion
  auto info = lm.minimize(coeff);
```

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp [hunk 2]
```diff
@@ -451,7 +451,7 @@ pcl::SampleConsensusModelEllipse3D<PointT>::optimizeModelCoefficients (
...
   for (Eigen::Index i = 0; i < coeff.size (); ++i)
     optimized_coefficients[i] = static_cast<float> (coeff[i]);
```
Comment (larshg) [inline sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp:457]:
```suggestion
  optimized_coefficients = coeff.cast<float>();
```

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp (hunk 1 above)
Comment (mvieth) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp:455]: Why do you propose to change this to auto? Yes, technically it is the enum `LevenbergMarquardtSpace::Status`, but further down in the print, the specifier `%i` expects an int, so casting to int here seems like the right choice.

Comment (larshg) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp:455]: Ahh, yes, then lets just keep it as is.

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp (hunk 2 above)
Comment (mvieth) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_ellipse3d.hpp]: Done

**Review by larshg — APPROVED** [review]

**Pull request merged**